stdlib/public/core/FlatMap.swift

```diff
@@ -67,4 +107,48 @@ extension LazyCollectionProtocol
   >> {
     return self.map(transform).flatten()
   }
+  
+  /// Returns a `LazyMapBidirectionalCollection` containing the concatenated non-nil
+  /// results of mapping transform over this collection.  The elements of
+  /// the result are computed lazily, each time they are read.
+  ///
+  /// Use this method to receive only nonoptional values when your
+  /// transformation produces an optional value.
+  ///
+  /// - Parameter transform: A closure that accepts an element of this
+  /// collection as its argument and returns an optional value.
+  public func flatMap<U>(
+    _ transform: (Self.Elements.Iterator.Element) -> U?
+  ) -> LazyMapBidirectionalCollection<
+    LazyFilterBidirectionalCollection<
+      LazyMapBidirectionalCollection<Self.Elements, U?>>,
+    U
+  > {
+    return self.map(transform).filter { $0 != nil }.map { $0! }
+  }
```

Comment:
The overloads for Sequence, Collection and BidirectionalCollection look very similar -- do you want to try gyb'ing them?